### PR TITLE
Add VisualDL logger support for Fabric and PyTorch Lightning

### PR DESCRIPTION
**Pull request by PlumBlossomMaid**
## What does this PR do?

This PR adds a new `VisualDLLogger` for both Fabric and PyTorch Lightning, enabling users to log metrics, images, audio, histograms, embeddings, PR/ROC curves, and more to [VisualDL](https://www.paddlepaddle.org.cn/paddle/visualdl) - a visualization analysis tool from PaddlePaddle.

### Key User Value
This PR enables PyTorch Lightning users to **seamlessly switch to VisualDL on [Baidu AI Studio](https://aistudio.baidu.com)** (a popular AI development platform in China), where user could use PyTorch but TensorBoard is not supported.

The `VisualDLLogger` follows the same API pattern as `TensorBoardLogger`, allowing users to replace it with **minimal code changes** (just change the logger class name) while getting full visualization support on AI Studio.

### Key Features
- **Scalar logging**: Track metrics with `log_metrics` and `log_dict`
- **Hyperparameter logging**: Save hparams to YAML and visualize in VisualDL
- **Image logging**: Support for numpy arrays, torch tensors, and file paths
- **Audio logging**: Support for numpy arrays, torch tensors, and file paths (via `soundfile`)
- **Histogram logging**: Track weight/gradient distributions
- **Text logging**: Log arbitrary text data
- **Embedding visualization**: Project high-dimensional data with metadata
- **PR/ROC curves**: For classification model evaluation
- **Automatic versioning**: Manages experiment versions like TensorBoard
- **Subdirectory support**: Organize logs with `sub_dir` parameter

### Background
[Baidu AI Studio](https://aistudio.baidu.com), a leading AI development platform in China, but does not support TensorBoard. 
This PR addresses the needs of Chinese developers by providing a fully compatible VisualDL logger,
allowing them to leverage Lightning's powerful features while using the visualization tool native to their platform.

### Implementation Details
- **Fabric version**: `lightning/fabric/loggers/visualdl.py` - core implementation
- **PyTorch version**: `lightning/pytorch/loggers/visualdl.py` - PyTorch Lightning integration
- **Tests**: Comprehensive test coverage (98% for PyTorch version, 79% for Fabric version)
- **Documentation**: Added usage guides for both Fabric and PyTorch Lightning

### Example Usage
**Fabric:**
```python
from lightning.fabric import Fabric
from lightning.fabric.loggers import VisualDLLogger

logger = VisualDLLogger(root_dir="logs", name="my_experiment")
fabric = Fabric(loggers=logger)
fabric.log("loss", loss)
```

**PyTorch Lightning:**
```python
from lightning.pytorch import Trainer
from lightning.pytorch.loggers import VisualDLLogger

logger = VisualDLLogger("logs", name="my_experiment")
trainer = Trainer(logger=logger)
```

### Related Issues
None

### Breaking Changes
None

----
📚 Documentation preview 📚: https://pytorch-lightning--21584.org.readthedocs.build/en/21584/

### CODE REVIEW

**Comment by williamFalcon** [comment]
we're not accepting new loggers. thanks though!
